### PR TITLE
[Validator] fix: Case-insensitive extensions in File-Constraint

### LINKED ISSUE #48859: [Validator] New extensions check of the File Constraint does not normalize the extensions
### Symfony version(s) affected

6.2.4

### Description

Hi, the [new `extensions` config for the File Constraint](https://symfony.com/doc/current/reference/constraints/File.html#extensions) does not normalize the checked extensions, nor the configured ones). So for example, when I allow the 'xml' extension and someone uploads a valid XML-file called 'MY.XML', that is considered as invalid.

As a workaround, I can also configure `XML` as a second allowed extension, but then someone will come and upload a 'My.Xml' next and so forth. Many smartphones name their camera files with uppercased extensions, so more often that should affect the `jpg` or `m4v` extensions.

### How to reproduce

Configure a File Constraint and set the `extensions` config:

````
new File(
    extensions: ['xml'],
),
````

Then upload an XML-file with a non-lowercase file-extension like `xml.XML` and observe that it is considered as a not allowed file-extension. Validate that by renaming it to `xml.xml`, upload it again and observe that it is considered as an allowed file-extension.

Or just have a look into the code of the [FileValidator.php](https://github.com/symfony/symfony/blob/6.3/src/Symfony/Component/Validator/Constraints/FileValidator.php) inside the block starting with `if ($constraint->extensions) {` to see how the check is currently implemented.

### Possible Solution

From my point of view, the check should instead of `if ($fileExtension !== $k)` do something like `if (mb_strtolower($fileExtension) !== mb_strtolower($k))` or normalize the checked (and configured) extensions in any other way so that the check is case-insensitive.

I could send a PR for that, if that would be an accepted solution.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by spackmat**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #48859 
| License       | MIT
| Doc PR        | none

This PR fixes the problem described in #48859 by normalizing the checked file-extension in the File Constraint to lowercase, when the `extensions` check is used. Previously a file with a `.TXT` extension was considered invalid, when the `.txt` extension was allowed by the Constraint. The test case was also updated.

### CODE REVIEW

File: src/Symfony/Component/Validator/Tests/Constraints/FileValidatorTest.php [hunk 1]
```diff
@@ -555,7 +555,7 @@ private function validExtensionProvider(): iterable
     {
         yield ['test.gif'];
         yield ['test.png.gif'];
-        yield ['ccc.txt'];
+        yield ['ccc.TXT'];
```
Comment (fancyweb) [inline src/Symfony/Component/Validator/Tests/Constraints/FileValidatorTest.php]: Thanks. You actually need to create the test fixture file `ccc.TXT`.

Comment (fancyweb) [reply src/Symfony/Component/Validator/Tests/Constraints/FileValidatorTest.php]: And use it (instead of replacing ccc.txt).

Comment (spackmat) [reply src/Symfony/Component/Validator/Tests/Constraints/FileValidatorTest.php]: okay, did so.

**Review by fancyweb — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @spackmat.

**Pull request merged**